File: ride-share-backend/check_db.py
import sqlite3

conn = sqlite3.connect('rideshare.db')
c = conn.cursor()

c.execute("SELECT name FROM sqlite_master WHERE type='table';")
print("Tables:", c.fetchall())

try:
    c.execute("SELECT vehicle_id, images FROM vehicle_images LIMIT 5")
    rows = c.fetchall()
    print(f"Found {len(rows)} image rows")
    for row in rows:
        vid = row[0]
        img = row[1]
        print(f"Vehicle: {vid}, Length: {len(img)}")
        if len(img) > 100:
            print(f"Start: {img[:50]} ... End: {img[-50:]}")
        else:
            print(f"Content: {img}")
except Exception as e:
    print("Error querying vehicle_images:", e)

conn.close()

File: ride-share-backend/check_db.js
const sqlite3 = require('sqlite3').verbose();
const db = new sqlite3.Database('./rideshare.db');

db.serialize(() => {
    db.all("SELECT * FROM vehicle_images", (err, rows) => {
        if (err) {
            console.error(err);
            return;
        }
        console.log(`Found ${rows.length} image rows`);
        if (rows.length > 0) {
            rows.slice(0, 3).forEach(row => {
                console.log(`Row: vehicle_id=${row.vehicle_id}, string length: ${row.images ? row.images.length : 0}`);
                if (row.images) {
                    console.log(`Starts with: ${row.images.substring(0, 100)}`);
                }
            });
        }
    });
});

db.close();
